app/src/main/java/org/wikipedia/page/PageFragment.java

```diff
@@ -919,43 +919,50 @@ void showFindInPage() {
-                = new FindInWebPageActionProvider(this, funnel);
 
-        startSupportActionMode(new ActionMode.Callback() {
-            @Override
-            public boolean onCreateActionMode(ActionMode mode, Menu menu) {
-                MenuItem menuItem = menu.add(R.string.menu_page_find_in_page);
-                menuItem.setActionProvider(findInPageActionProvider);
-                menuItem.expandActionView();
-                setToolbarElevationEnabled(false);
-                return true;
+        bridge.evaluate(JavaScriptActionHandler.expandCollapsedTables(true), msg -> {
+            if (!isAdded()) {
+                return;
             }
+            final FindInPageFunnel funnel = new FindInPageFunnel(app, model.getTitle().getWikiSite(),
+                    model.getPage() != null ? model.getPage().getPageProperties().getPageId() : -1);
+            final FindInWebPageActionProvider findInPageActionProvider
+                    = new FindInWebPageActionProvider(this, funnel);
+
+            startSupportActionMode(new ActionMode.Callback() {
+                @Override
+                public boolean onCreateActionMode(ActionMode mode, Menu menu) {
+                    MenuItem menuItem = menu.add(R.string.menu_page_find_in_page);
+                    menuItem.setActionProvider(findInPageActionProvider);
+                    menuItem.expandActionView();
+                    setToolbarElevationEnabled(false);
+                    return true;
+                }
 
-            @Override
-            public boolean onPrepareActionMode(ActionMode mode, Menu menu) {
-                mode.setTag("actionModeFindInPage");
-                return false;
-            }
+                @Override
+                public boolean onPrepareActionMode(ActionMode mode, Menu menu) {
+                    mode.setTag("actionModeFindInPage");
+                    return false;
+                }
 
-            @Override
-            public boolean onActionItemClicked(ActionMode mode, MenuItem item) {
-                return false;
-            }
+                @Override
+                public boolean onActionItemClicked(ActionMode mode, MenuItem item) {
+                    return false;
+                }
 
-            @Override
-            public void onDestroyActionMode(ActionMode mode) {
-                if (webView == null || !isAdded()) {
-                    return;
+                @Override
+                public void onDestroyActionMode(ActionMode mode) {
+                    if (webView == null || !isAdded()) {
+                        return;
+                    }
+                    funnel.setPageHeight(webView.getContentHeight());
+                    funnel.logDone();
+                    webView.clearMatches();
+                    hideSoftKeyboard();
+                    setToolbarElevationEnabled(true);
+                    bridge.execute(JavaScriptActionHandler.expandCollapsedTables(false));
```

Comment:
I'm not sure we need to re-close the tables when this is done.
Suppose I open the find-in-page box, and it finds the text I'm looking for _inside_ a table. Then I exit out of find-in-page, but the table suddenly closes.

Reply:
Makes sense to me. Removed it.